### PR TITLE
Only show argument completion items when method accepts arguments

### PR DESCRIPTION
**Pull request by genlu**
Address https://github.com/dotnet/vscode-csharp/issues/6374

### CODE REVIEW

**Review by dibarbet — APPROVED** [review]

**Comment by genlu** [comment]
@dibarbet FYI I needed to add a workaround for a bug in type inferer

File: src/Features/CSharp/Portable/Completion/CompletionProviders/SymbolCompletionProvider.cs [hunk 1]
```diff
@@ -86,20 +86,51 @@ protected override string GetFilterText(ISymbol symbol, string displayText, CSha
...
+            CompletionOptions options,
+            CancellationToken cancellationToken)
+        {
+            if (completionContext is null)
+                return true;
+
+            // If we are triggered in argument list, provide symbols only when the invoked method accept any arguments.
+            // Special case for argument of base record primary constructor as a workaround for bug https://github.com/dotnet/roslyn/issues/70803
+            return !await IsTriggeredInArgumentListAsync(completionContext, position, options, cancellationToken).ConfigureAwait(false)
+                || (!context.InferredTypes.IsEmpty || IsInPrimaryConstructorArgumentList());
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/Completion/CompletionProviders/SymbolCompletionProvider.cs:111]: this is so confusing.  please break into separate clauses (and doc them).

File: src/Features/CSharp/Portable/Completion/CompletionProviders/SymbolCompletionProvider.cs [hunk 2]
```diff
@@ -86,20 +86,51 @@ protected override string GetFilterText(ISymbol symbol, string displayText, CSha
...
+
+            bool IsInPrimaryConstructorArgumentList()
+                => context.TargetToken.Parent?.Parent?.IsKind(SyntaxKind.PrimaryConstructorBaseType) is true;
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/Completion/CompletionProviders/SymbolCompletionProvider.cs:114]: this is a bit of an odd expression given the name of hte local function.  IsIn implies that we're *anywhere* within the arg list.  but here it seems like it's asking about very specific topmost nodes in the arg list.

File: src/Features/CSharp/Portable/Completion/CompletionProviders/SymbolCompletionProvider.cs [hunk 3]
```diff
@@ -86,20 +86,51 @@ protected override string GetFilterText(ISymbol symbol, string displayText, CSha
             CompletionOptions options,
             CancellationToken cancellationToken)
         {
-            if (context != null && options.TriggerInArgumentLists)
+            if (context is null)
+                return true;
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/Completion/CompletionProviders/SymbolCompletionProvider.cs:90]: when is the context null?

Comment (genlu) [reply src/Features/CSharp/Portable/Completion/CompletionProviders/SymbolCompletionProvider.cs:90]: Here's one location I found
https://github.com/dotnet/roslyn/blob/a35b73595c954a0281fac24e0b6136eca572db7a/src/Features/Core/Portable/Completion/Providers/AbstractRecommendationServiceBasedCompletionProvider.cs#L214

File: src/Features/CSharp/Portable/Completion/CompletionProviders/SymbolCompletionProvider.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/Features/CSharp/Portable/Completion/CompletionProviders/SymbolCompletionProvider.cs:114]: my point is that the impl of this method doesn't match the name :)

File: src/Features/CSharp/Portable/Completion/CompletionProviders/SymbolCompletionProvider.cs (hunk 3 above)
Comment (CyrusNajmabadi) [reply src/Features/CSharp/Portable/Completion/CompletionProviders/SymbolCompletionProvider.cs:90]: thanks!

**Pull request merged**